### DISCUSSION TITLE
Making wd/hub parameterised and optional

### DISCUSSION THREAD

**Discussion by AnilPatidar** [AppiumTestDistribution/AppiumTestDistribution#1093] [Ideas]
@saikrishna321 When i was running script using cloud, and specifically pClould. I was getting error as no routes found. Because some how Http request were having prefix as wd/hub  and remaining url . 

Wd/hub/session/{sessionid}/click 
no routes found 

line of code to be modified 

                     withArgument(GeneralServerFlag.BASEPATH, "/wd/hub") In local appium services class

@saikrishna321 if you feel, it’s a good idea, i will raise a PR

**Comment by AnilPatidar** [comment]
1. We might ignore this wd/hub if it is cloud 
2. Or we might take this as a param

**Reply by saikrishna321** [reply]
Happy to review a PR. Lets go with option 2.

**Reply by AnilPatidar** [reply]
@saikrishna321 is this good idea to take this as a system property ? 
if it is passed , we can set ,else it will be ignored.
mvn clean -Dtest=Runner -Dbasepath=wd/hub test

**Reply by saikrishna321** [reply]
@AnilPatidar We can also add this to caps.json. 

Currently we have appiumServerPath

We can add appiumBasePath

**Reply by AnilPatidar** [reply]
sure
